components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/util/DCRMUtils.java

```diff
@@ -122,7 +123,11 @@ public static DCRMClientException generateClientException(DCRMConstants.ErrorMes
     public static boolean isRegexValidated(String applicationName) {
 
         String spValidatorRegex = DCRConstants.APP_NAME_VALIDATING_REGEX;
-        return isRegexValidated(applicationName, spValidatorRegex);
+
+        if (spNameRegexPattern == null) {
+            spNameRegexPattern = Pattern.compile(spValidatorRegex);
```

Comment:
Assuming 'spValidatorRegex' is constant, can assign this at initialization time.